ci: test_detect_calculator_has_uia fails — Calculator only reports vision, not UIA

## Problem
The integration test `test_detect_calculator_has_uia` fails because Calculator's detection chain only returns `['vision']` instead of including `'uia'`.

## Steps to Reproduce
1. Run `python -m pytest tests/integration/test_unified_app_model.py::TestDetectionChainCalculator::test_detect_calculator_has_uia -v`
2. Test asserts `'uia' in method_names` but gets `['vision']`

## Expected
Calculator should be detected with UIA support (UWP app with UIA automation).

## Actual
```
AssertionError: Calculator should support UIA, got methods: ['vision']
assert 'uia' in ['vision']
```

## Environment
- Windows 11 Pro 10.0.26200
- naturo v0.3.1
- Runner: NATUROBOT
- Commit: 701243e

## Notes
This may be related to Calculator's UWP framework detection. The app is running and detectable (vision works), but UIA detection is failing.

**[QA-Mariana]**

--- a/naturo/process.py
+++ b/naturo/process.py
@@ -361,6 +361,9 @@ def launch_app(
 
     cmd_args = args or []
     system = platform.system()
+    # Set when launching via cmd /c start — triggers PID resolution (#785)
+    _resolve_real_pid: str | None = None
+    _resolve_real_alias: str | None = None
 
     try:
         if system == "Windows":
@@ -406,6 +409,11 @@ def launch_app(
                     # Launched but already exited — report success with a dummy PID
                     return ProcessInfo(pid=0, name=name or "", path="", is_running=False)
                 proc = subprocess.Popen(["cmd", "/c", "start", "", resolved_name] + cmd_args)
+                # cmd.exe exits quickly after launching the target app.
+                # Mark for real PID resolution below — proc.pid is cmd.exe,
+                # not the actual application (#785).
+                _resolve_real_pid = resolved_name
+                _resolve_real_alias = name
         elif system == "Darwin":
             if path:
                 proc = subprocess.Popen([path] + cmd_args)
@@ -448,15 +456,43 @@ def launch_app(
     except OSError as exc:
         raise AppNotFoundError(launch_target, suggested_action=str(exc))
 
+    # Resolve real PID when launched via cmd /c start (#785).
+    # cmd.exe exits after spawning the target app; proc.pid is the wrapper,
+    # not the actual application (especially for UWP apps like Calculator).
+    real_pid = proc.pid
+    real_name = name or os.path.basename(path or "")
+    real_path = path or ""
+    if _resolve_real_pid is not None:
+        try:
+            proc.wait(timeout=5)
+        except subprocess.TimeoutExpired:
+            pass
+        # Poll for the real process — give UWP apps time to start
+        poll_deadline = time.monotonic() + (timeout if wait_until_ready else 3.0)
+        while time.monotonic() < poll_deadline:
+            found = find_process(name=_resolve_real_pid)
+            if not found and _resolve_real_alias and _resolve_real_alias != _resolve_real_pid:
+                found = find_process(name=_resolve_real_alias)
+            if found:
+                real_pid = found.pid
+                real_name = found.name
+                real_path = found.path or ""
+                break
+            time.sleep(0.3)
+
     info = ProcessInfo(
-        pid=proc.pid,
-        name=name or os.path.basename(path or ""),
-        path=path or "",
+        pid=real_pid,
+        name=real_name,
+        path=real_path,
         is_running=True,
         window_count=0,
     )
 
     if wait_until_ready:
+        # If we already resolved the real PID via find_process (#785),
+        # the app is confirmed running — skip the poll loop.
+        if _resolve_real_pid is not None and real_pid != proc.pid:
+            return info
         start = time.monotonic()
         while time.monotonic() - start < timeout:
             time.sleep(0.5)

--- a/tests/test_process.py
+++ b/tests/test_process.py
@@ -434,6 +434,119 @@ class TestLaunchApp:
             launch_app(name="app", wait_until_ready=True, timeout=2.0)
 
 
+@patch(
+    "naturo.cli.interaction._is_current_session_interactive",
+    return_value=True,
+)
+class TestLaunchAppPidResolution:
+    """Tests for real PID resolution after cmd /c start (#785).
+
+    On Windows, launch_app uses ``cmd /c start`` which returns cmd.exe's PID,
+    not the actual application.  These tests verify that the real app PID is
+    resolved via find_process.
+    """
+
+    @staticmethod
+    def _make_run_result(returncode=0, stdout="", stderr=""):
+        import subprocess as _sp
+        return _sp.CompletedProcess(args=[], returncode=returncode, stdout=stdout, stderr=stderr)
+
+    @patch("naturo.process.platform.system", return_value="Windows")
+    @patch("naturo.process.find_process")
+    @patch("naturo.process.subprocess.run")
+    @patch("naturo.process.subprocess.Popen")
+    def test_resolves_real_pid_after_cmd_start(
+        self, mock_popen, mock_run, mock_find, _mock_platform, _mock_session,
+    ):
+        """When find_process locates the real app, its PID replaces cmd.exe's."""
+        mock_proc = MagicMock()
+        mock_proc.pid = 100  # cmd.exe PID
+        mock_proc.wait.return_value = 0
+        mock_popen.return_value = mock_proc
+        mock_run.return_value = self._make_run_result(returncode=0, stdout="C:\\notepad.exe")
+        mock_find.return_value = ProcessInfo(pid=5678, name="notepad.exe", path="C:\\notepad.exe")
+
+        info = launch_app(name="notepad")
+        assert info.pid == 5678
+        assert info.name == "notepad.exe"
+
+    @patch("naturo.process.platform.system", return_value="Windows")
+    @patch("naturo.process.find_process")
+    @patch("naturo.process.subprocess.run")
+    @patch("naturo.process.subprocess.Popen")
+    def test_falls_back_to_cmd_pid_when_process_not_found(
+        self, mock_popen, mock_run, mock_find, _mock_platform, _mock_session,
+    ):
+        """When find_process returns None, falls back to cmd.exe's PID."""
+        mock_proc = MagicMock()
+        mock_proc.pid = 100
+        mock_proc.wait.return_value = 0
+        mock_popen.return_value = mock_proc
+        mock_run.return_value = self._make_run_result(returncode=0, stdout="C:\\notepad.exe")
+        mock_find.return_value = None
+
+        info = launch_app(name="notepad", timeout=0.5)
+        assert info.pid == 100  # Fallback to cmd.exe PID
+
+    @patch("naturo.process.platform.system", return_value="Windows")
+    @patch("naturo.process.find_process")
+    @patch("naturo.process.subprocess.run")
+    @patch("naturo.process.subprocess.Popen")
+    def test_resolves_via_alias_fallback(
+        self, mock_popen, mock_run, mock_find, _mock_platform, _mock_session,
+    ):
+        """When the resolved name doesn't match, tries the original name."""
+        mock_proc = MagicMock()
+        mock_proc.pid = 100
+        mock_proc.wait.return_value = 0
+        mock_popen.return_value = mock_proc
+        mock_run.return_value = self._make_run_result(returncode=0, stdout="C:\\calc.exe")
+
+        # First call (resolved_name) returns None, second call (original name) finds it
+        mock_find.side_effect = [None, ProcessInfo(pid=7777, name="calc.exe", path="C:\\calc.exe")]
+
+        info = launch_app(name="calculator")
+        assert info.pid == 7777
+
+    @patch("naturo.process.platform.system", return_value="Windows")
+    @patch("naturo.process.find_process")
+    @patch("naturo.process.subprocess.run")
+    @patch("naturo.process.subprocess.Popen")
+    def test_wait_until_ready_skips_poll_when_resolved(
+        self, mock_popen, mock_run, mock_find, _mock_platform, _mock_session,
+    ):
+        """wait_until_ready returns immediately when real PID is already found."""
+        mock_proc = MagicMock()
+        mock_proc.pid = 100
+        mock_proc.wait.return_value = 0
+        mock_proc.poll.return_value = 0  # cmd.exe already exited
+        mock_popen.return_value = mock_proc
+        mock_run.return_value = self._make_run_result(returncode=0, stdout="C:\\app.exe")
+        mock_find.return_value = ProcessInfo(pid=9999, name="app.exe")
+
+        info = launch_app(name="app", wait_until_ready=True, timeout=1.0)
+        assert info.pid == 9999
+        assert info.is_running is True
+
+    @patch("naturo.process.platform.system", return_value="Windows")
+    @patch("naturo.process.find_process", return_value=None)
+    @patch("naturo.process.subprocess.run")
+    @patch("naturo.process.subprocess.Popen")
+    def test_path_launch_skips_pid_resolution(
+        self, mock_popen, mock_run, mock_find, _mock_platform, _mock_session,
+    ):
+        """Launching by --path uses Popen directly, no PID resolution needed."""
+        mock_proc = MagicMock()
+        mock_proc.pid = 4444
+        mock_proc.wait.return_value = 0
+        mock_popen.return_value = mock_proc
+
+        with patch("os.path.isfile", return_value=True):
+            info = launch_app(path="C:\\notepad.exe")
+        assert info.pid == 4444
+        mock_find.assert_not_called()
+
+
 class TestResolveLaunchName:
     """Tests for _resolve_launch_name alias resolution (#246)."""
 

